frontend/src/breeding_form.tsx

```diff
@@ -0,0 +1,199 @@
+/**
+ * @file This file contains the BreedingForm function. This function allows
+ *       users to create and update breeding events in the database.
+ */
+import React from 'react'
+import { makeStyles } from '@material-ui/core/styles';
+import { MuiPickersUtilsProvider, KeyboardDatePicker
+        } from '@material-ui/pickers'
+import DateFnsUtils from '@date-io/date-fns'
+import { dateFormat, Genebank, individualLabel, inputVariant, LimitedHerd,
+        OptionType } from '@app/data_context_global';
+
+import { Breeding } from './breeding_list';
+import { TextField, Typography } from '@material-ui/core';
+import { useDataContext } from './data_context';
+import { Autocomplete } from '@material-ui/lab';
+
+const useStyles = makeStyles({
+  form: {
+    width: '100%',
+    height: '100%',
+    overflow: 'hidden',
+    padding: '20px',
+  },
+  formBox: {
+    border: '1px solid lightgrey',
+    borderRadius: '8px',
+    padding: '10px',
+  },
+  wideControl: {
+    width: '100%',
+  },
+});
+
+const emptyBreeding: Breeding = {
+  breed_date: '',
+  breed_notes: '',
+  father: '',
+  mother: '',
+  birth_date: '',
+  birth_notes: '',
+  litter_size: 0,
+}
+
+/**
+ * The BreedingForm function. This function allows users to create and update
+ * breeding events in the database.
+ */
+export function BreedingForm({data, herdId}: {data: Breeding | 'new', herdId: string | undefined}) {
+  const style = useStyles();
+  const { genebanks } = useDataContext()
+  const [form, setForm] = React.useState({...emptyBreeding} as Breeding)
+  const hasBirth = React.useMemo(() => {return !!form.birth_date}, [form])
```

Comment:
```suggestion
  const hasBirth = !!form.birth_date
```
This is quick enough to not memo